### PR TITLE
feature(lxy): add value rescale

### PR DESCRIPTION
**Pull request by karroyan**
## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/policy/dqn.py [hunk 1]
```diff
@@ -197,15 +207,29 @@ def _forward_learn(self, data: Dict[str, Any]) -> Dict[str, Any]:
...
+            loss, td_error_per_sample = q_nstep_td_error(data_n, self._gamma, nstep=self._nstep, value_gamma=value_gamma)
+            
 
         # ====================
         # Q-learning update
         # ====================
         self._optimizer.zero_grad()
         loss.backward()
+        if self._cfg.other.reward_scale.reward_scale_function == 'consgrad':
+            grad_v_norm = self.g_v(data['next_obs'])
```
Comment (PaParaZz1) [inline ding/policy/dqn.py:226]: call `g_v` behind td error will lead to gradient mixture, your grad_v here is `grad_v + grad_td`

File: ding/policy/dqn.py [hunk 2]
```diff
@@ -197,15 +207,29 @@ def _forward_learn(self, data: Dict[str, Any]) -> Dict[str, Any]:
...
+        else:
+            loss, td_error_per_sample = q_nstep_td_error(data_n, self._gamma, nstep=self._nstep, value_gamma=value_gamma)
+            
 
         # ====================
         # Q-learning update
         # ====================
         self._optimizer.zero_grad()
         loss.backward()
+        if self._cfg.other.reward_scale.reward_scale_function == 'consgrad':
```
Comment (PaParaZz1) [inline ding/policy/dqn.py:225]: this method will remove target network in DQN

File: ding/policy/dqn.py [hunk 3]
```diff
@@ -652,3 +684,21 @@ def _load_state_dict_learn(self, state_dict: Dict[str, Any]) -> None:
...
+
+class PopartUpdate(object):
+    def __init__(self, shape):
+        self._shape = shape
+        self._mean = torch.zeros(self._shape)
+        self._var = torch.ones(self._shape)
+
+    def update(self, value, W, b):
+        batch_mean = torch.mean(value, axis=0) # dim = 6
+        batch_var = torch.var(value, axis=0)
```
Comment (PaParaZz1) [inline ding/policy/dqn.py:696]: you should use std rather than var

File: ding/policy/dqn.py [hunk 4]
```diff
@@ -652,3 +684,21 @@ def _load_state_dict_learn(self, state_dict: Dict[str, Any]) -> None:
...
+        W_new = torch.reshape(1/batch_var * self._var, (self._shape, 1))
+        W_new = W * W_new
+
+        b_new = 1/batch_var *(self._var*b + self._mean - batch_mean)
+        self._var = batch_var
+        self._mean = batch_mean
```
Comment (PaParaZz1) [inline ding/policy/dqn.py:702]: where is your EMA update for mean and std

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]
